internal/handlers/validator/custom.go

```diff
@@ -99,6 +99,20 @@ func uuidValidator(fl validator.FieldLevel) bool {
 	return val != uuid.UUID{}
 }
 
+func labelValidator(fl validator.FieldLevel) bool {
+	val, ok := fl.Field().Interface().(string)
+	if !ok {
+		return false
+	}
+
+	// Label key/value should not be empty
+	// Allow alphanumeric characters, hyphens, underscores, and dots
+	// Must start and end with alphanumeric character
+	labelRegex := regexp.MustCompile(`^[a-zA-Z0-9]([a-zA-Z0-9._-]*[a-zA-Z0-9])?$`)
```

Comment:
It's better to run this once, not and each time the function is called. So would be better to extract it module var.